### ISSUE TITLE
-rescan doesn't work, how to get transactions?

### ISSUE THREAD

**Issue by skironDotNet** [firoorg/firo#1145]
I have wallet created using mnemonic, but first ever funds are sent to address 30 in BIP order before wallet file created. 
I created wallet and it was scanning but didn't detect any transactions. Is it because BIP rules "check first 20 addresses if no balance assume no more scanning needed?" if BIP rule is applied then this is wrong, BIP rule is for SPV wallets like Coinomi, TrustWallet etc.

I added new addresses using Receiving addresses window, until address 40, so I would assume all I need is to run wallet with -rescan flag but this does nothing.
Another hack I tried is to importprivkey with true to rescan, because in other wallets that invokes scanning and grabs all other addresses balances too, but here can't do it because wallet created using mnemonic is very strict

> Importing wallets and private keys is disabled for mnemonic-enabled wallets.To import your dump file, create a non-mnemonic wallet by setting "usemnemonic=0" in your firo.conf file, after backing up and removing your existing wallet. (code -4)

How to force firod to rescan for missing transactions?

I also tried `-rescan -zapwallettxes=1`

**Comment by skironDotNet** [comment]
While a bug exists that -rescan does nothing, I've found a reason why wallet scan during initial creation from mnemonic didn't find any transactions. In my config I had set `keypool=0` for wallet engineering reasons (dumpwallet, importwallet, wanted to have as less "initial" addresses as possible in non-mnemonic wallet). 
Having `keypool=0` technically creates only 1 address + 1 change address in the wallet, so because my funds start from address 30, it wasn't scanning anything past address 1 during initial wallet creation from mnemonic (so it was not "BIP rule" - answering myself :))

Can you fix `-rescan` to actually force rescan using all available addresses in the wallet that may have been "added" later than the transaction happened?

**Comment by justanwar** [comment]
Setting `keypool=0` overrides `-rescan`'s default value of 100 keys. It is not recommended to set a low keypool number with `-rescan` (higher than default is okay).

**Comment by skironDotNet** [comment]
yes, but even with keypool=0 you can add addresses later, and -rescan should check all wallet addresses on start and it doesn't, so keypool=0 was just a problem during wallet creation, the bug is still valid that -rescan doesn't work

**Comment by levonpetrosyan93** [comment]
First of all it is not technically possible to check all possible addresses, that is why there is specified number  for keypool size, which is 100 by default, so it creates 100 keys and checks the balances for that 100 addresses, if it finds used addresses, it generates more. 
It is against logic to force wallet to create 1 address, which is not used, and and run rescan to restore balance from 30-th address. Please recover your wallet from mnemonics and put bigger number for kepool size or leave it to use default number.

**Comment by skironDotNet** [comment]
Just close the ticket if you don't understand what am I talking about. I admitted about key pool being a reason for not seeing funds, but do this, generate mnemonic using this https://iancoleman.io/bip39/
use testnet, or if you have funds to spent use mainnet, keep your keypool at 100 send the funds to 102 once funds confirmed restore the wallet from mnemonic. Now obviously won't see your funds by design, happy?

Go to receiving addresses window and add 103 more so you "initialize" an address in your wallet (keypool 100 + 3 = 103) This won't be scanned by design as expected, great. Now how to get the balance to appear in the wallet?

>   -rescan
       Rescan the block chain for missing wallet transactions on startup

It works in all coins I tried so far, to name a few, DGB, PIVX, LTC, whatever, but in FIRO it does not, and that's the bug.
Rescan should take all addresses in the wallet and re-scan the chain transactions and detect any balances in them.
